packages/shared/src/hooks/post/useViewPost.ts

```diff
@@ -0,0 +1,62 @@
+import {
+  UseMutateAsyncFunction,
+  useMutation,
+  useQueryClient,
+} from '@tanstack/react-query';
+import { sendViewPost } from '../../graphql/posts';
+import { generateQueryKey, RequestKey } from '../../lib/query';
+import { useAuthContext } from '../../contexts/AuthContext';
+import { ReadingDay, UserStreak } from '../../graphql/users';
+import { getTodayTz } from '../../lib/dateFormat';
+
+export const useViewPost = (): UseMutateAsyncFunction<
+  unknown,
+  unknown,
+  string
+> => {
+  const client = useQueryClient();
+  const { user } = useAuthContext();
+  const streakKey = generateQueryKey(RequestKey.UserStreak, user);
+  const readKey = generateQueryKey(RequestKey.ReadingStreak30Days, user);
+  const { mutateAsync: onSendViewPost } = useMutation(sendViewPost, {
+    onSuccess: async () => {
+      await client.invalidateQueries(streakKey);
+      client.setQueryData<UserStreak>(streakKey, (data) => {
+        const streak = { ...data };
+
+        if (
+          !streak?.lastViewAt ||
+          new Date(streak.lastViewAt).getDate() !== new Date().getDate()
+        ) {
+          streak.max += 1;
+          streak.total += 1;
+          streak.current += 1;
+        }
+
+        streak.lastViewAt = new Date();
+
+        return streak;
+      });
+
+      const reading = client.getQueryData<ReadingDay[]>(readKey);
+
+      if (reading) {
+        const timezoned = getTodayTz(user?.timezone || 'UTC');
+        const date = new Date(timezoned).toISOString().split('T')[0];
```

Comment:
Can't we directly do `timezoned.toISOString().split('T')[0];` ? Seems like what the above function return is already a `new Date(timeZonedToday)`

As mentioned above, do we have anything against using the date libraries (datefns)? Definitely think it can be safer to use

Reply:
This is what we use in our API, so I think it is better to match it.

> Can't we directly do timezoned.toISOString().split('T')[0]; ? Seems like what the above function return is already a new Date(timeZonedToday)

Oh, right. Yeah, we can do that.